testScripts/ObjectDownloadMedia.js

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable no-console */
+const ScriptBase = require("./parentClasses/ScriptBase");
+const utils = require("../src/Utils");
+const Fraction = require("fraction.js");
+const {JSONPath} = require("jsonpath-plus");
+const fs = require("fs");
+const path = require("path");
+
+class ObjectDownloadMedia extends ScriptBase {
+  async body() {
+    const client = await this.client();
+
+    let objectId = this.args.objectId;
+    const versionHash = this.args.objectHash;
+    const offeringType = this.args.offeringType || "default";
+    const startTime = this.args.startTime;
+    const endTime = this.args.endTime;
+    const out = this.args.out || "./out";
+
+    if(objectId === undefined){
+      if(versionHash===undefined){
+        throw Error("require object-id or object-hash to be provided");
+      }
+      const res = utils.DecodeVersionHash(versionHash);
+      objectId = res.objectId;
+    }
+
+    const dirPath = path.resolve(out);
+    if (!fs.existsSync(dirPath)) {
+      fs.mkdirSync(dirPath, { recursive: true });
+      console.log(`Directory created at ${dirPath}`);
+    } else {
+      throw Error(`Directory already exists at ${dirPath}`);
+    }
+
+
+    const sourcesJsonPath = "offerings." + offeringType + ".media_struct.streams.video.sources[*]";
```

Comment:
You may want to add an option `--streamKey` that defaults to "video", and then statement would be `"offerings." + offeringType + ".media_struct.streams." + streamKey + ".sources[*]";`

Video stream is not guaranteed to be stored under the key "video", and (maybe?) a  future user might want to download an audio track